### PR TITLE
Markdown/Rich text enhancements [FEATURE/FIX]

### PR DESCRIPTION
**Pull request by bryhasagithub**
Related and resolves a papercut issue found in  Glide repo.
https://github.com/quicktype/glide/issues/10185
Uploading Screen Recording 2021-09-02 at 4.42.56 PM.mov…

### Details:
- Adds an icon for accepting changes when modifying markdown text in a cell
- Adds additional padding to prevent any vertical scroll bars from interfering with the icons.

### CODE REVIEW

File: src/data-grid-overlay-editor/private/markdown-overlay-editor.tsx [hunk 1]
```diff
@@ -20,11 +20,25 @@ const MarkdownOverlayEditor: React.FunctionComponent<Props> = p => {
     const [editMode, setEditMode] = React.useState<boolean>(markdown === "" || forceEditMode);
 
     const onEditClick = React.useCallback(() => {
-        setEditMode(true);
-    }, []);
+        setEditMode(!editMode);
+    }, [editMode]);
```
Comment (ivoelbert) [inline src/data-grid-overlay-editor/private/markdown-overlay-editor.tsx:24]:
```ts
const onEditClick = React.useCallback(() => {
    setEditMode(e => !e);
}, []);
```

File: src/data-grid-overlay-editor/private/markdown-overlay-editor.tsx [hunk 2]
```diff
@@ -20,11 +20,25 @@ const MarkdownOverlayEditor: React.FunctionComponent<Props> = p => {
...
+                    style={{ marginTop: 10 }}
+                    autoFocus={true}
+                    onKeyDown={onKeyDown}
+                    value={markdown}
+                    onChange={onChange}
+                />
+                <div className="edit-icon" onClick={onEditClick}>
+                    <Checkmark />
+                </div>
+            </MarkdownOverlayEditorStyle>
```
Comment (ivoelbert) [inline src/data-grid-overlay-editor/private/markdown-overlay-editor.tsx:40]: why the `style` tags? 👀

File: src/growing-entry/growing-entry.tsx [hunk 3]
```diff
@@ -44,5 +45,5 @@ const GrowingEntryImpl: React.FunctionComponent<Props> = (props: Props) => {
     );
 };
 
-const GrowingEntry = React.memo(GrowingEntryImpl);
+const GrowingEntry = GrowingEntryImpl;
```
Comment (ivoelbert) [inline src/growing-entry/growing-entry.tsx:48]: Just export the `GrowingEntry` and get rid of the export default ❤️

**Review by ivoelbert — COMMENTED** [review]
Leaving a few drive-by comments

**Comment by bryhasagithub** [comment]
"Drive-by" changes before Design review. There are some things to clarify in regards to behavior in use cases that I tried to show in the video: 

https://user-images.githubusercontent.com/11837954/135517849-65632527-f520-455a-ba3b-00456134be1a.mov

**Comment by bryhasagithub** [comment]
These are with the new updates and changes Tristan suggested
https://user-images.githubusercontent.com/11837954/135542600-f376f6c7-159b-4f9a-87c4-b8ba6b23be8d.mov

**Pull request merged**